### PR TITLE
Adding the ability to pass code directly to lambda layer_version

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #591

@lukehoban we should have a chat about this before we merge it - I am not quite sure that this has no unintended side-effects

### CODE REVIEW

File: examples/lambda-layer/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+let lambda = new aws.lambda.LayerVersion("my-layer-version", {
+    code: new asset.AssetArchive({
+        "index.js": new asset.StringAsset(
+            "exports.handler = (e, c, cb) => cb(null, {statusCode: 200, body: 'Hello, world!'});",
+        ),
+    }),
+    compatibleRuntimes: [
+        aws.lambda.NodeJS8d10Runtime,
+        aws.lambda.NodeJS8d10Runtime,
```
Comment (lukehoban) [inline examples/lambda-layer/index.ts]: Should these two be different?

File: examples/lambda-layer/index.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    ],
+    layerName: "lambda_layer_name",
+});
```
Comment (lukehoban) [inline examples/lambda-layer/index.ts]: Is the goal of the test to create a lambda function using this layer?  Or just to create the Layer resource?

File: examples/lambda-layer/invoke.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+echo "Enter your Lambda's name (usually in the form mylambda-f0780fd95f4) followed by [ENTER]: "
+read LAMBDA
+echo "Attempting to invoke $LAMBDA ..."
+aws lambda invoke --function-name $LAMBDA --log-type Tail out.txt | jq '.LogResult' -r | base64 --decode
```
Comment (lukehoban) [inline examples/lambda-layer/invoke.sh]: I don't think we use this file anymore - it was a very old artifact of back when these really were "examples".  And for the Layer in particular, this won't even work.

File: examples/lambda-layer/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "serverless-raw",
```
Comment (lukehoban) [inline examples/lambda-layer/package.json]: Nit - but this name probably should change.

File: examples/lambda-layer/index.ts (hunk 1 above)
Comment (stack72) [reply examples/lambda-layer/index.ts]: yes - good spot!

File: examples/lambda-layer/invoke.sh (hunk 3 above)
Comment (stack72) [reply examples/lambda-layer/invoke.sh]: gone - good spot!

File: examples/lambda-layer/package.json (hunk 4 above)
Comment (stack72) [reply examples/lambda-layer/package.json]: done!

File: examples/lambda-layer/index.ts (hunk 2 above)
Comment (stack72) [reply examples/lambda-layer/index.ts]: The goal of the test is just to make sure that we can create the layer based on the new parameters - I have added some outputs to verify that the layer has been created as expected

**Comment by stack72** [comment]
@CyrusNajmabadi FYI here ☝️

**Pull request merged**